spec/models/verified_principal_spec.rb

```diff
@@ -0,0 +1,50 @@
+RSpec.describe VerifiedPrincipal do
+  describe '#register!' do
+    subject{ described_class.new(form_data, 'ABC') }
+    
+    let(:user_params) { 
+      {
+        email: 'test@maps.org.uk',
+        password: 'Password1*',
+        password_confirmation: 'Password1*'
+      }
+    }
+    let(:principal_params) {
+      {
+        fca_number: '123456',
+        first_name: 'Margo',
+        last_name: 'Test',
+        email_address: 'test@maps.org.uk',
+        job_title: 'Retirement Adviser',
+        telephone_number: '2075554343',
+        confirmed_disclaimer: 1
+      }
+    }
+    let(:principal) { double(Principal) }
+    let(:form_data) { principal_params.merge(user_params) }
+    let(:form) { NewPrincipalForm.new(form_data) }
+    let(:user) { double(User, principal: Principal.new) }
+    let(:identification) { double(Identification) }
+
+    before do
+      allow(form).to receive(:user_params).and_return(user_params)
+      allow(form).to receive(:principal_params).and_return(principal_params)
+    end
+
+    it 'creates the user, principal and associated firm' do
+      expect(NewPrincipalForm).to receive(:new).and_return(form)
+      expect(User).to receive(:new).with(user_params).and_return(user)
+      expect(user).to receive(:build_principal).with(principal_params)
+      expect(user).to receive(:save!)
+      expect(Stats).to receive(:increment).with('radsignup.principal.created')
+      expect(Identification)
+        .to receive(:contact)
+        .with(user.principal)
+        .and_return(identification)
+      expect(identification).to receive(:deliver_later)
+      expect(NewFirmMailer).to receive_message_chain(:notify, :deliver_later)
+
+      subject.register!
```

Comment:
There are something about this test that sounds like a smell to me, I think there are too much setup and stubs. I don't have any suggestion but shall we pair on this to try to refactor the method?

Reply:
As I talked in person, I don't have any suggestion about this. I will left to the author take the decision on the path to take about this. 🙏

Reply:
I'm of 2 minds about the comment. I understand why you have made the comment about the `code smell` due to the test setup, but I also am not sure if refactoring will provide any benefit at this stage. It may be that as we use the api more, we find a better case for refactoring. Since there has been no suggestion about how to refactor or follow up on pairing, I'm going to leave it as it is.